controller_manager/src/controller_manager.cpp

```diff
@@ -2057,7 +2057,9 @@ controller_interface::return_type ControllerManager::update(
 
       if (controller_go)
       {
-        auto controller_ret = loaded_controller.c->update(time, controller_period);
+        const auto controller_actual_period =
+          (time - *loaded_controller.next_update_cycle_time) + controller_period;
```

Comment:
Shouldn't this be something like `time-time_old_`?

Reply:
The `next_update_cycle_time` is computed such that there is a condition that checks if the time is greater than or equal to the current time and if the condition passes, then it triggers the update. Usually, when it is equal this difference is zero, but that means you have a control period in between.  

If not, I can check if I can move some logic around to see if I can avoid adding this controller period at this stage. I will check this evening.

Reply:
Ok, maybe this is fine as it is.